src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -0,0 +1,174 @@
+import { createExistingTx } from '@/services/tx/txSender'
+import { txDispatch, TxEvent } from '@/services/tx/txEvents'
+import { generateDataRowValue } from '@/components/transactions/TxDetails/Summary/TxDataRow'
+import { Errors, logError } from '@/services/exceptions'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import extractTxInfo from '@/services/tx/extractTxInfo'
+import { BatchExecuteData } from '@/components/tx/modals/BatchExecuteModal/index'
+import DecodedTxs from '@/components/tx/modals/BatchExecuteModal/DecodedTxs'
+
+const getSignatures = (confirmations: Record<string, string>) => {
+  return Object.entries(confirmations)
+    .filter(([_, signature]) => Boolean(signature))
+    .sort(([signerA, _], [signerB, __]) => signerA.toLowerCase().localeCompare(signerB.toLowerCase()))
+    .reduce((prev, [_, signature]) => {
+      return prev + signature.slice(2)
+    }, '0x')
+}
+
+const getMultiSendTxs = (
+  txs: TransactionDetails[],
+  chain: ChainInfo,
+  safeAddress: string,
+  safeVersion: string,
+): MetaTransactionData[] => {
+  const safeContractInstance = getGnosisSafeContractInstance(chain, safeVersion)
+
+  return txs
+    .map((tx) => {
+      if (!isMultisigExecutionDetails(tx.detailedExecutionInfo)) return
+
+      const args = extractTxInfo(tx, safeAddress)
+      const sigs = getSignatures(args.signatures)
+
+      const data = safeContractInstance.interface.encodeFunctionData('execTransaction', [
+        args.txParams.to,
+        args.txParams.value,
+        args.txParams.data,
+        args.txParams.operation,
+        args.txParams.safeTxGas,
+        args.txParams.baseGas,
+        args.txParams.gasPrice,
+        args.txParams.gasToken,
+        args.txParams.refundReceiver,
+        sigs,
+      ])
+
+      return {
+        operation: OperationType.Call,
+        to: safeAddress,
+        value: '0',
+        data,
+      }
+    })
+    .filter(Boolean) as MetaTransactionData[]
+}
+
+const getTxsWithDetails = (txs: Transaction[], chainId: string) => {
+  return Promise.all(
+    txs.map(async (tx) => {
+      return await getTransactionDetails(chainId, tx.transaction.id)
+    }),
+  )
+}
+
+const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubmit: (data: null) => void }) => {
+  const [multiSendContract, setMultiSendContract] = useState<Multi_send_call_only>()
+  const [multiSendTxData, setMultiSendTxData] = useState<string>()
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+  const chain = useCurrentChain()
+  const { safe } = useSafeInfo()
+  const provider = useWeb3()
+
+  const [txsWithDetails, error, loading] = useAsync<TransactionDetails[]>(() => {
+    if (!chain?.chainId) return
+
+    return getTxsWithDetails(data.txs, chain.chainId)
+  }, [data.txs, chain?.chainId])
+
+  useEffect(() => {
+    if (!chain) return
+
+    setMultiSendContract(getMultiSendCallOnlyContractInstance(chain.chainId))
```

Comment:
Would we benefit in moving the instance to an `ExternalStore` with an intiation hook? Or is it just used here?